cmake/zephyr_module.cmake

```
@@ -59,6 +59,32 @@ if(WEST OR ZEPHYR_MODULES)
     endforeach()
   endif()
 
+  if(EXISTS ${CMAKE_BINARY_DIR}/zephyr_modules.txt)
+    file(STRINGS ${CMAKE_BINARY_DIR}/zephyr_modules.txt ZEPHYR_MODULES_TXT
+         ENCODING UTF-8)
+    set(ZEPHYR_MODULE_NAMES)
+
+    foreach(module ${ZEPHYR_MODULES_TXT})
+      # Match "<name>":"<path>" for each line of file, each corresponding to
+      # one module. The use of quotes is required due to CMake not supporting
+      # lazy regexes (it supports greedy only).
+      string(REGEX REPLACE "\"(.*)\":\".*\":\".*\"" "\\1" module_name ${module})
+      string(REGEX REPLACE "\".*\":\"(.*)\":\".*\"" "\\1" module_path ${module})
+      string(REGEX REPLACE "\".*\":\".*\":\"(.*)\"" "\\1" cmake_path ${module})
+
+      list(APPEND ZEPHYR_MODULE_NAMES ${module_name})
+
+      string(TOUPPER ${module_name} MODULE_NAME_UPPER)
+      if(NOT ${MODULE_NAME_UPPER} STREQUAL CURRENT)
+        set(ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR ${cmake_path})
```

I rebased and retested the new version and it looks like a typo snuck in here. This should be:

```suggestion
        set(ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR ${module_path})
```

---

Good catch, thanks.

Wonder how that sneaked in, cause it is correct here: https://github.com/zephyrproject-rtos/zephyr/pull/27740/commits/67c09760e740f9872275592c99d23ffec7d52036
Guess i'll never know.